independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/ClassPathElement.java

```diff
@@ -144,6 +144,10 @@ public void close() {
     };
 
     default List<ClassPathResource> getResources(String name) {
-        return List.of(getResource(name));
+        ClassPathResource resource = getResource(name);
+        if (resource != null) {
```

Comment:
Right I think this is necessary. We might want to clarify the javadoc on `getResource`: I see many other uses are implementing the nullcheck, so they expect it, but the contract could be clarified.